server/utils.go

```diff
@@ -63,7 +63,11 @@ func SendHTTPRequest(ctx context.Context, client http.Client, method, url string
 	if err != nil {
 		return 0, err
 	}
-	defer resp.Body.Close()
+	defer func() {
+		// drain body completely to reuse connection
+		_, _ = io.Copy(io.Discard, resp.Body)
```

Comment:
do you have any data on a lack of reuse of connections?

if you look below the `io.ReadAll` utility is used to consume the response body so according to the docs, the `Close` (as currently exists) is sufficient to trigger the reuse mechanism

Reply:
Thanks Alex, would be curious too.